Add ability to retry intermittent tasks

gregarndt opened the pull request.

gregarndt commented:
Missing right now is the schema, I'll add that soon.

walac commented on test/integration/retry_intermittent_task_test.js:
```diff
@@ -0,0 +1,70 @@
+    });
+
+    assert(result.run.state === 'pending', 'new pending run should be created');
+    assert(result.run.reasonCreated === 'task-retry', 'run should have been created becaus eof task retry');
```
nit: "because of"

walac commented on test/integration/retry_intermittent_task_test.js:
```diff
@@ -0,0 +1,70 @@
+
+    assert(result.run.state === 'completed', 'task should not fail or be retried');
+    assert(result.run.reasonResolved === 'completed', 'task should not or be retried');
+    assert(result.status.runs.length === 1);
```
replace `assert` by `assert.equal` for better error reporting.

walac commented:
Some tests failing...

gregarndt commented:
Thanks for the comments @walac .  I've fixed up those assertions and added a schema.  Any tests that fail now should hopefully just be our normal run of the mill intermittent failures and we can land this.

gregarndt commented:
Thank you sir!
